src/nupic/engine/__init__.py

```diff
@@ -28,7 +28,11 @@
 import functools
 
 basicTypes = ['Byte', 'Int16', 'UInt16', 'Int32', 'UInt32', 'Int64', 'UInt64',
-              'Real32', 'Real64', 'Handle']
+              'Real32', 'Real64', 'Handle', 'Bool']
+
+arrayTypes = ['ByteArray', 'Int16Array', 'UInt16Array', 'Int32Array',
+              'UInt32Array', 'Int64Array', 'UInt64Array', 'Real32Array',
+              'Real64Array', 'BoolArray']
```

Comment:
Good point, this is a little sloppy. I pushed an update, let me know what you think.